src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -16,6 +17,10 @@ abstract class LazyModule()(implicit val p: Parameters)
   protected[diplomacy] var info: SourceInfo = UnlocatableSourceInfo
   protected[diplomacy] val parent = LazyModule.scope
 
+
+  var childrenAop = ()=>children
+  var nodesAop = ()=>nodes
```

Comment:
This pattern of a `var` and `() => ...` is strange. Should you use a `lazy val` instead? Why does it need to be mutable? Also, please add a comment to describe the reason for this change.

Reply:
suggest we call this "childrenAOP" if AOP is an acronym.